x/dex/keeper/msg_server.go

```diff
@@ -26,6 +27,10 @@ func (k MsgServer) Deposit(
 	goCtx context.Context,
 	msg *types.MsgDeposit,
 ) (*types.MsgDepositResponse, error) {
+	if err := msg.ValidateBasic(); err != nil {
+		return nil, errors.Wrap(err, "failed to validate MsgDeposit")
+	}
+
```

Comment:
in the task description it is said that
> Now the logic is deprecated and msg.ValidateBasic() is not being called automatically anymore.

but in the docs a different statement is made:
<img width="671" alt="image" src="https://github.com/neutron-org/neutron/assets/34917380/5795a63a-828b-48ec-aac6-8001055d7471">

so, is ValidateBasic called twice now? once in the BaseApp, once here in Deposit?

Reply:
This is a good catch, i missed that paragraph.

Cosmos SDK still [calls](https://github.com/cosmos/cosmos-sdk/blob/c4d9a495052b78a02723ed1df1d336efe83a4e8d/baseapp/baseapp.go#L620) method if a struct implements `HasValidateBasic` interface.

I think we can rename all `ValidateBasic` methods to `Validate` to remove the implementation and thus the method will be called only once in our MsgServers

Reply:
All ValidateBasic() module message methods have been renamed to Validate().